src/main/java/com/snowflake/kafka/connector/SnowflakeSinkTask.java

```diff
@@ -243,6 +248,16 @@ public void close(final Collection<TopicPartition> partitions) {
    */
   @Override
   public void put(final Collection<SinkRecord> records) {
+    if (enableRebalancing && records.size() > 0) {
+      rebalancingCounter++;
+      if (rebalancingCounter == rebalancingThreshold) {
+        try {
+          Thread.sleep(310000);
+        } catch (InterruptedException e) {
+          e.printStackTrace();
+        }
+      }
+    }
```

Comment:
let's extract this out of put and keep the current put() clean of any testing code if possible.